app/quest-boost/claim/[boostId]/page.tsx

```diff
@@ -0,0 +1,194 @@
+"use client";
+
+import React, { useEffect, useMemo, useState } from "react";
+import styles from "@styles/questboost.module.css";
+import { getBoostById, getQuestBoostClaimParams } from "@services/apiService";
+import { useAccount } from "@starknet-react/core";
+import Button from "@components/UI/button";
+import { useNotificationManager } from "@hooks/useNotificationManager";
+import { NotificationType, TransactionType } from "@constants/notifications";
+import { CDNImage } from "@components/cdn/image";
+import Lottie from "lottie-react";
+import claimWinnerLottie from "@public/visuals/sq_claim.json";
+import claimLoserLottie from "@public/visuals/sq_looser.json";
+import { hexToDecimal } from "@utils/feltService";
+import { boostClaimCall } from "@utils/callData";
+import { getTokenName } from "@utils/tokenService";
+import useBoost from "@hooks/useBoost";
+
+type BoostQuestPageProps = {
+  params: {
+    boostId: string;
+  };
+};
+
+export default function Page({ params }: BoostQuestPageProps) {
+  const { address, account } = useAccount();
+  const { boostId } = params;
+  const [boost, setBoost] = useState<Boost>();
+  const [sign, setSign] = useState<Signature>(["", ""]);
+  const { addTransaction } = useNotificationManager();
+  const [displayCard, setDisplayCard] = useState<boolean>(false);
+  const [displayLottie, setDisplayLottie] = useState<boolean>(true);
+  const [transactionHash, setTransactionHash] = useState<string>("");
+  const { updateBoostClaimStatus } = useBoost();
+
+  const fetchPageData = async () => {
+    const boostInfo = await getBoostById(boostId);
+    setBoost(boostInfo);
+  };
+
+  useEffect(() => {
+    fetchPageData();
+  }, []);
+
+  const fetchBoostClaimParams = async (): Promise<Signature> => {
+    let formattedSign: Signature = ["", ""];
+    try {
+      if (!boost?.id || !address) return formattedSign;
+      const res = await getQuestBoostClaimParams(boost.id);
+      formattedSign = [res?.r, res?.s];
+      return formattedSign;
+    } catch (err) {
+      console.log(err);
+      return formattedSign;
+    }
+  };
+
+  const handleClaimClick = async () => {
+    if (isUserWinner) {
+      const signature = await fetchBoostClaimParams();
+      if (!signature) return;
+      setSign(signature);
+    } else {
+      updateBoostClaimStatus(parseInt(boostId), false);
+    }
+  };
+
+  useEffect(() => {
+    if (!account || !boost || sign[0].length === 0 || sign[1].length === 0)
+      return;
+
+    const callContract = async () => {
+      const { transaction_hash } = await account.execute(
+        boostClaimCall(boost, sign)
+      );
+      updateBoostClaimStatus(boost?.id, true);
+      setTransactionHash(transaction_hash);
+    };
+
+    callContract();
+  }, [sign]);
```

Comment:
why only `sign` here? don't you have other dependencies on the `useEffect` ? normally your linter should tell you this.

Reply:
added